### DISCUSSION TITLE
Unable to Connect-PnPOnline using secret with no authentication

### DISCUSSION THREAD

**Discussion by 200mpg** [pnp/powershell#4121] [Q&A]
Setup secret store, added secret name and value, configured store for no authentication. Run the below command:

![image](https://github.com/user-attachments/assets/67ba1a9c-27a9-45e0-8b30-95a5c9e7249c)

Receive the below:

![image](https://github.com/user-attachments/assets/1fa8727d-fb2f-4486-8cf6-437981e0169f)

**Comment by mluteran** [comment]
im having the same issue

**Comment by ToddKlindt** [comment]
What command are you using the create the secret? It looks like it's being stored as the type Secure String. Connect-PnPOnline is looking for a PSCredential object or a string there. If the object you pass it isn't a string or PSCredential then it can't use it. 
Unless you're married to using an external secret store, you can use the Windows Credential Manager. It's much easier. Use [Add-PnPStoredCredential](https://pnp.github.io/powershell/cmdlets/Add-PnPStoredCredential.html) to store the credentials. If you use the url for the name (https://contoso.sharepoint.com) then Connect-PnPOnline will pick it up automatically. Or, you can specify its name with the -Credentials parameter. I do this for all of my credentials and it works really well.

**Reply by 200mpg** [reply]
Thanks for the reply!

I am using a local vault to store the secret that was generated by the app in Azure. We need to be able to store the secret string securely, then pass it through the cmdlet in order to automate via task. The documentation suggests this is the primary way to authenticate for automation:

https://pnp.github.io/powershell/articles/credentialmanagement.html

![image](https://github.com/user-attachments/assets/7edc9f07-1e47-48ba-98ce-7de53ef34f04)

**Comment by ToddKlindt** [comment]
You don't need any of that for automation. This is how I do it (assuming no MFA)
Run this to reset the default secret store to CredMan (Windows Credential Manager)
`Set-SecretVaultDefault -Name CredMan`
Then use this to set the stored credential you want to use:
`Add-PnPStoredCredential -Name https://contoso.sharepoint.com -Username admin@contoso.sharepoint.com`
It will prompt you for your password.
Now you have a couple of options. You can retrieve the PSCredential object like this:
`$Creds = Get-PnPStoredCredential -Name https://contoso.sharepoint.com`
Then pass $Creds to the -Credentials parameter of Connect-PnPOnline. The beauty of this is that you can use $Creds with anything, like the GraphSDK, the Exchange module, whatever.
Or, you can also just run this:
`Connect-PnPOnline -Url https://contoso.sharepoint.com`
Since no credentials were passed Connect-PnPOnline will look in the default credential store for a matching entry. Since you named your entry https://contoso.sharepoint.com it will match and Connect-PnPOnline will use it.

**Reply by 200mpg** [reply]
Unfortunately, our sec team will only allow API creds at this time for these apps when using service accounts. The secret-vault process in the pnp powershell documentation looks to be the primary way they have enabled that capability.

**Comment by 200mpg** [comment]
I understand there isnt an SLA for this product, but its frustrating to be blocked by step 1
